3.9.1/index.d.ts

```diff
@@ -57,8 +23,23 @@ declare interface Gulp {
    * </ul>
    */
   task: typeof Orchestrator.prototype.add;
-  run: RunMethod;
-  watch: typeof watch;
+  /**
+   * Run the specified task(s)
+   * @deprecated will be removed in gulp 4.0. Use task dependencies instead.
+   */
+  run: typeof Orchestrator.prototype.start;
+  /**
+   * Run the specified task(s)
+   * @deprecated will be removed in gulp 4.0. Use task dependencies instead.
+   */
+  start: typeof Orchestrator.prototype.start;
+  /**
+   * watches all files in the specified glob(s) for changes and runs the given task(s).
+   * @param globs
+   * @param tasks
+   */
+  watch(globs?: string | Array<string>, tasks?: task | Array<task>): EventEmitter;
+  watch(globs?: string | Array<string>, opt?: WatchOptions, tasks?: task | Array<task>): EventEmitter;
```

Comment:
@felixfbecker When I followed up on the types for the return type I came to the conclusion that it is an EventEmitter.
And on the array desctiption topic: It doesn't make sense as I see it. Writing x[] looks like some abstract syntax-only thing to define an array (as you see in PHP or Java). But in ECMAScript that's not the case. Array is an actual class and by writing the Array<x> notation you express that it's an object of type Array that contains "somethings" (may be instances of objects or primitives) of type x. Beyond that it is often easier in more complex type notations to use generic notation because in short notation you might end up needing braces in the types which makes it harder to understand.
TL;DR: I don't really see the point in writing x[] rather than Array<x>. Can you give me objective reasons for that notation?